llvm/lib/CodeGen/SelectionDAG/DAGCombiner.cpp

```diff
@@ -25465,6 +25498,22 @@ SDValue DAGCombiner::visitCONCAT_VECTORS(SDNode *N) {
   return SDValue();
 }
 
+SDValue DAGCombiner::visitVECTOR_INTERLEAVE(SDNode *N) {
+  // Check to see if all operands are identical.
+  if (!llvm::all_equal(N->op_values()))
+    return SDValue();
+
+  // Check to see if the identical operand is a splat.
+  SDValue Splat = DAG.getSplatValue(N->getOperand(0));
+  if (!Splat)
```

Comment:
Sorry, I should have noticed this before but does `if (!isSplatValue(N->getOperand(0))` work here?